apps/total-typescript/src/trpc/routers/purchases.ts

```diff
@@ -0,0 +1,54 @@
+import {getSdk, Purchase} from '@skillrecordings/database'
+import {publicProcedure, router} from '@skillrecordings/skill-lesson'
+import {get, isEmpty, last} from 'lodash'
+import {getToken} from 'next-auth/jwt'
+import {z} from 'zod'
+
+export const purchasesRouter = router({
+  getLastPurchase: publicProcedure.query(async ({ctx}) => {
```

Comment:
This query is being used on TotalTypeScript to get `bulkCoupon.id` required for the SelfRedeem component.